AdyenDropIn/Presentation/ModalViewController.swift

```diff
@@ -92,20 +96,41 @@ internal final class ModalViewController: UIViewController {
     
     internal lazy var separator: UIView = {
         let separator = UIView(frame: .zero)
-        separator.backgroundColor = style.separatorColor ?? UIColor.Adyen.componentSeparator
+        separator.backgroundColor = getSeparatorColor()
         return separator
     }()
     
-    internal lazy var toolbar: UIView = {
-        let toolbar = ModalToolbar(title: self.innerController.title,
-                                   style: style,
-                                   cancelHandler: { [weak self] in self?.didCancel() })
-        toolbar.translatesAutoresizingMaskIntoConstraints = false
-        return toolbar
+    private func getSeparatorColor() -> UIColor {
+        switch navBarType {
+        case .regular:
+            return style.separatorColor ?? UIColor.Adyen.componentSeparator
+        case .custom:
+            return style.backgroundColor
+        }
+    }
+    
+    internal lazy var navBar: UIView = {
+        let navBar: UIView
+        
+        switch navBarType {
+        case .regular:
+            navBar = getRegularNavBar()
+        case .custom(let customNavbar):
+            navBar = customNavbar
+        }
+        
+        navBar.translatesAutoresizingMaskIntoConstraints = false
+        return navBar
     }()
     
+    private func getRegularNavBar() -> UIView {
+        ModalToolbar(title: self.innerController.title,
+                     style: style,
+                     cancelHandler: { [weak self] in self?.didCancel() })
```

Comment:
we can assign call of `didCancel` to AnyNavigationBar.cancelHandler 

```swift
navBar.cancelHandler = { [weak self] in self?.didCancel() })
```

Reply:
good idea